stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/utils/AndroidUtils.kt

```diff
@@ -162,3 +170,33 @@ inline fun <T> safeCall(default: T, block: () -> T): T {
         default
     }
 }
+
+/**
+ * Start a foreground service with a service type to meet requirements introduced in Android 14.
+ *
+ * @param notificationId The notification ID
+ * @param notification The notification to show
+ * @param trigger The trigger that started the service: [TRIGGER_ONGOING_CALL], [TRIGGER_OUTGOING_CALL], [TRIGGER_INCOMING_CALL], [TRIGGER_SHARE_SCREEN]
+ */
+internal fun Service.startForegroundWithServiceType(
+    notificationId: Int,
+    notification: Notification,
+    trigger: String,
+) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+        val foregroundServiceType = when (trigger) {
+            TRIGGER_ONGOING_CALL -> ServiceInfo.FOREGROUND_SERVICE_TYPE_PHONE_CALL
+            TRIGGER_OUTGOING_CALL, TRIGGER_INCOMING_CALL -> ServiceInfo.FOREGROUND_SERVICE_TYPE_SHORT_SERVICE
+            TRIGGER_SHARE_SCREEN -> ServiceInfo.FOREGROUND_SERVICE_TYPE_MEDIA_PROJECTION
+            else -> ServiceInfo.FOREGROUND_SERVICE_TYPE_SHORT_SERVICE
+        }
+        ServiceCompat.startForeground(
+            this,
+            notificationId,
+            notification,
+            foregroundServiceType,
+        )
+    } else {
+        startForeground(notificationId, notification)
+    }
+}
```

Comment:
I believe we can just use the `ServiceCompat.startForeground` function instead of separating it into using `startForeground,` as `ServiceCompat.startForeground` already implemented the details depending on the Android version.

```kotlin
    public static void startForeground(@NonNull Service service, int id,
            @NonNull Notification notification, int foregroundServiceType) {
        if (Build.VERSION.SDK_INT >= 34) {
            Api34Impl.startForeground(service, id, notification, foregroundServiceType);
        } else if (Build.VERSION.SDK_INT >= 29) {
            Api29Impl.startForeground(service, id, notification, foregroundServiceType);
        } else {
            service.startForeground(id, notification);
        }
    }
```

So following the internal implementation, we can pass the `foregroundServiceType` after S(`Build.VERSION.SDK_INT >= 29`) and we also need to handle the `ForegroundServiceStartNotAllowedException` following the [official documentation](https://developer.android.com/develop/background-work/services/foreground-services#start).

Reply:
At first I intended to just use `ServiceCompat`, judging by it's name, but I got a lint error in AS. But probably I missed something. Good catch, I'll have a second look!

Reply:
Hey @liviu-timar can we update this to only use the `ServiceCompat` after which we can merge the PR.

Reply:
Yep, will do it soon. 👍

Reply:
Did the changes, ready for merging with the caveat that I'm not sure that the `PHONE_CALL` type is appropriate, as per [docs](https://developer.android.com/about/versions/14/changes/fgs-types-required#phone-call).

Reply:
We will update this separately by providing new service types.